src/MiniProfiler.Shared/Internal/Render.cs

```diff
@@ -82,6 +82,10 @@ public static class Render
             {
                 sb.Append(" data-start-hidden=\"true\"");
             }
+            if (renderOptions?.Nonce.HasValue() ?? false)
+            {
+                sb.Append(" nonce=\"").Append(renderOptions.Nonce).Append("\"");
```

Comment:
Since I see that `sb` contains other markup elements, (line 114) it is presumed that it's contents are `HtmlSafe`? Is there any concern that `Nonce` is an XSS point, or at least requires some escaping?

Alternatively, what about simply rendering the incorrect thing? If a nonce is supposed to be random and someone's nonce generator uses the printable ascii range (0x20-0x7F), `&`, `"`, etc. could feasibly end up breaking rendering if this isn't escaping like I am concerned about.

It is assumed that the caller will do the escaping for miniprofiler

Reply:
Doh, absolutely - this was the first text thing a user can pass and definitely needs encoding, total goof. Updated (and added a few tests for these cases).